FGEGraphics/ClientSystem/ShaderLocations.cs

```diff
@@ -42,4 +46,309 @@ public static class Common2D
+
+        /// <summary>
+        /// The matrices of the light sources (essentially their point of view this is used for shadow mapping).
+        /// Uniform: shadow_matrix_array
+        /// </summary>
+        public const int SHADOW_MATRIX_ARRAY = 10;
+    }
+
+    /// <summary>Locations used in the hdrpass shader.</summary>
+    public static class HDRPass
+    {
+        /// <summary>
+        /// The screen size.
+        /// Uniform: u_screen_size
+        /// </summary>
+        public const int SCREEN_SIZE = 4;
+    }
+
+    /// <summary>Locations used in the godray shader.</summary>
+    public static class Godray
+    {
+        /// <summary>
+        /// The amount of exposure to apply.
+        /// Uniform: exposure
+        /// </summary>
+        public const int EXPOSURE = 6;
+
+        /// <summary>
+        /// The aspect ratio of the screen ensures the godray effect is properly scaled.
+        /// Uniform: aspect
+        /// </summary>
+        public const int ASPECT_RATIO = 7;
+
+        /// <summary>
+        /// The position of the light source (sun location).
+        /// Uniform: sunloc
+        /// </summary>
+        public const int SUN_LOCATION = 8;
+
+        /// <summary>
+        /// The density of the godray effect.
+        /// Higher values make the godray effect more dense and focused while lower values make it more spread out.
+        /// Uniform: density
+        /// </summary>
+        public const int DENSITY = 12;
+
+        /// <summary>
+        /// The minimum depth used for depth linearization.
+        /// Uniform: MIN_DEPTH
+        /// </summary>
+        public const int MIN_DEPTH = 14;
+
+        /// <summary>
+        /// The maximum depth used for depth linearization.
+        /// Uniform: MAX_DEPTH
+        /// </summary>
+        public const int MAX_DEPTH = 15;
+
+        /// <summary>
+        /// Distance that determines whether a pixel is in the sky or not.
+        /// Uniform: SKY_DIST
+        /// </summary>
+        public const int SKY_DISTANCE = 16;
+    }
+
+    /// <summary>Locations used in the finalgodray shader.</summary>
+    public static class FinalGodray
+    {
+        /// <summary>
+        /// What position the camera is targeting in the world (ray traced).
+        /// Uniform: cameraTargetPos
+        /// </summary>
+        public const int CAMERA_TARGET_POSITION = 8;
+
+        /// <summary>
+        /// How far away the camera target position is from the camera.
+        /// Uniform: cameraTargetDepth
+        /// </summary>
+        public const int CAMERA_TARGET_DEPTH = 9;
+
+        /// <summary>
+        /// The HDR exposure value.
+        /// Uniform: hdrExposure
+        /// </summary>
+        public const int HDR_EXPOSURE = 10;
+
+        /// <summary>
+        /// Camera position relative to rendering origin.
+        /// Uniform: cameraPos
+        /// </summary>
+        public const int CAMERA_RELATIVE_POSITION = 14;
```

Comment:
`relative` is redundant here, _everything_ in the graphics end is relative to the rendering origin, naturally

Reply:
I just followed what the comment said in the shader.